lib/rubygems/update_suggestion.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+##
+# Mixin methods for Gem::Command to promote available RubyGems update
+
+module Gem::UpdateSuggestion
+  # list taken from https://github.com/watson/ci-info/blob/7a3c30d/index.js#L56-L66
+  CI_ENV_VARS = [
+    "CI", # Travis CI, CircleCI, Cirrus CI, Gitlab CI, Appveyor, CodeShip, dsari
+    "CONTINUOUS_INTEGRATION", # Travis CI, Cirrus CI
+    "BUILD_NUMBER", # Jenkins, TeamCity
+    "CI_APP_ID", "CI_BUILD_ID", "CI_BUILD_NUMBER", # Applfow
+    "RUN_ID" # TaskCluster, dsari
```

Comment:
Interesting!

I think we should leverage these ENV vars here: https://github.com/rubygems/rubygems/blob/7de840c32e4a98f885613e6953b1a6eada6f48a3/bundler/lib/bundler/fetcher.rb#L223-L236 so that we can improve a CI env detection at https://ecosystem.rubytogether.org!

Reply:
I can somehow merge that list. I was thinking to keep this functionality in RubyGems itself if possible and from Bundler just check if this functionality is present and reuse as much as possible. But I'm afraid I'll need to keep this list in Bundler only for the original purpose just to not mess with metrics.

Reply:
I think it makes sense to merge both lists and keep it in RubyGems, although as usual, if we want Bundler accuracy when reporting this to be the same across RubyGems versions, then we'll need to backport it to Bundler and keep the list duplicated in Bundler for a while.

Reply:
I have not tackled this yet, my plan is to provide next PR to make same "call to update" feature for `bundle install`. I'll try to share as much as possible and consolidate those 2 CI env lists. For example looking randomly at first 3 entries, they all support `CI` env var as well. We can probably compact this list. Thinking about this again, I think I can do that in separated PR actually.

https://docs.travis-ci.com/user/environment-variables/#default-environment-variables
https://circleci.com/docs/built-in-environment-variables/
https://docs.semaphoreci.com/ci-cd-environment/environment-variables/

Reply:
Sounds good! I created https://github.com/rubygems/rubygems/issues/6038 to keep track of that.